### PR TITLE
Allow usage of GDAL's virtual file systems

### PR DESCRIPTION
**Pull request by jhkennedy**
**This is a companion PR to leiyangleon/autoRIFT#16**

---

Instead of modifying `http://...` paths with `/vsicurl/`, this instead:
* directly passes provided paths through to GDAL, allowing use of any [GDAL virtual file system driver](https://gdal.org/user/virtual_file_systems.html). Users would provide paths like:
  * local: `/path/to/file/...`
  * remote via url: `/vsicurl/https://...` 
  * remote on S3 (allows access to requester pays buckets): `/vsis3/...`
  * etc.
* This however does mean that one cannot assume files are co-registered, so co-registration is always performed.

Also moves main driver inside a `main()` function, and does:
```python
if __name__ == '__main__':
   main()

```
So that the test scripts can turned into [entrypoints](https://packaging.python.org/specifications/entry-points/) if vendored downstream.

---

This also:
* fixes the regex for Landsat-8 Collection-2 Level-1 products
* restricts the regex for Sentinel-2 and  Landsat-8  to look for expected file prefixes
* allows passing in and out needed parameters instead of autoRIFT relying on `fgrep`

### CODE REVIEW

File: testGeogrid_ISCE.py [hunk 1]
```diff
@@ -279,22 +273,46 @@ def runGeogrid(info, info1, dem, dhdx, dhdy, vx, vy, srx, sry, csminx, csminy, c
...
+        'ycount': None,  # FIXME: Get from C object (is calculated) or another source
+        'dt': obj.repeatTime,
+        'epsg': kwargs.get('epsg'),
+        'XPixelSize': None,  # FIXME: Get from C object (is calculated) or another source
+        'YPixelSize': None,  # FIXME: Get from C object (is calculated) or another source
+        'pixsizex': None,  # FIXME: Get from C object (is calculated) or another source
+        'rangePixelSize': None,  # FIXME: Get from C object (is calculated) or another source
+        'azimuthPixelSize': None,  # FIXME: Get from C object (is calculated) or another source
+    }
+
```
Comment (jhkennedy) [inline testGeogrid_ISCE.py:266]: I haven't done the work to figure out how to get these parameters in and out of the C++ object, or get them from another location...

File: testGeogrid_ISCE.py [hunk 2]
```diff
@@ -144,13 +144,14 @@ def loadMetadata(indir):
 
 def loadMetadataOptical(indir):
```
Comment (leiyangleon) [inline testGeogrid_ISCE.py]: In the main function, you now use **coregisterLoadMetadataOptical** for optical imagery. It seems this function **loadMetadataOptical** is deprecated. Do you still want to include it or just get rid of it?

File: testGeogrid_ISCE.py [hunk 3]
```diff
@@ -191,6 +190,7 @@ def coregisterLoadMetadataOptical(indir_m, indir_s):
     import struct
     import re
 
+    import isce
```
Comment (leiyangleon) [inline testGeogrid_ISCE.py:156]: Does excluding this import give you error message? I usually do not import the whole isce library and it worked well.

Comment (jhkennedy) [reply testGeogrid_ISCE.py:156]: ISCE  does these shenanigans on import:
https://github.com/isce-framework/isce2/blob/main/__init__.py#L41-L44

So, unless  users have done that to their environment manually, you need to `import isce` before trying to import anything inside those paths. Conda install of `isce`, for example, expect you to `import isce` first.

File: testGeogrid_ISCE.py (hunk 2 above)
Comment (jhkennedy) [reply testGeogrid_ISCE.py]: either way works for me -- I'd lean remove it.

File: testGeogridOptical.py [hunk 4]
```diff
@@ -225,18 +219,36 @@ def runGeogrid(info, info1, dem, dhdx, dhdy, vx, vy, srx, sry, csminx, csminy, c
 
     obj.runGeogrid()
 
-
-if __name__ == '__main__':
+    run_info = {
+        'chipsizex0': obj.chipSizeX0,
+        'vxname': vx,
+        'vyname': vy,
+        'sxname': srx,
```
Comment (leiyangleon) [inline testGeogridOptical.py]: this should be `'sxname': kwargs.get('dhdxs')`

`srx` is search range; while `dhdxs` is the smoothed slope file (see the previous line: `sxname = str.split(runCmd('fgrep "Slopes:" testGeogrid.txt'))[1][:-4]+"s.tif"`)

File: testGeogridOptical.py [hunk 5]
```diff
@@ -225,18 +219,36 @@ def runGeogrid(info, info1, dem, dhdx, dhdy, vx, vy, srx, sry, csminx, csminy, c
...
+        'syname': sry,
```
Comment (leiyangleon) [inline testGeogridOptical.py]: Similarly, this should be `'syname': kwargs.get('dhdys')`

**Comment by leiyangleon** [comment]
Please fix my minor comments. LGTM otherwise. Will merge it then.

File: testGeogridOptical.py (hunk 4 above)
Comment (jhkennedy) [reply testGeogridOptical.py]: Good catch! Done.

File: testGeogridOptical.py (hunk 5 above)
Comment (jhkennedy) [reply testGeogridOptical.py]: Good catch! Done.

File: testGeogrid_ISCE.py (hunk 2 above)
Comment (jhkennedy) [reply testGeogrid_ISCE.py]: Removed

File: testGeogrid_ISCE.py [hunk 6]
```diff
@@ -279,22 +243,47 @@ def runGeogrid(info, info1, dem, dhdx, dhdy, vx, vy, srx, sry, csminx, csminy, c
...
+        'yoff': None,  # FIXME: Get from C object (is calculated) or another source
+        'xcount': None,  # FIXME: Get from C object (is calculated) or another source
+        'ycount': None,  # FIXME: Get from C object (is calculated) or another source
+        'dt': obj.repeatTime,
+        'epsg': kwargs.get('epsg'),
+        'XPixelSize': None,  # FIXME: Get from C object (is calculated) or another source
+        'YPixelSize': None,  # FIXME: Get from C object (is calculated) or another source
+        'pixsizex': None,  # FIXME: Get from C object (is calculated) or another source
+        'rangePixelSize': None,  # FIXME: Get from C object (is calculated) or another source
+        'azimuthPixelSize': None,  # FIXME: Get from C object (is calculated) or another source
```
Comment (leiyangleon) [inline testGeogrid_ISCE.py]: Please remove these as they are not necessary. `rangePixelSize` is same as `XPixelSize` above and similar for y. `pixelsizex` is defined in testautoRIFT and is equivalent to `XPixelSize` as well. In other words, we should try to simplify runinfo definition and make it look the same between radar and optical although specific terms could have their own (and slightly different) meanings. 

I will fix the rest of the 'FIXME's.

Comment (jhkennedy) [reply testGeogrid_ISCE.py]: Done!

**Pull request merged**